src/CanRender.tsx

```diff
@@ -1,5 +1,12 @@
-import React from 'react';
-import { ICanRenderProps } from './interfaces/ICanRenderProps';
-import { CanRenderBase } from './CanRenderBase';
+import React, { Suspense } from 'react';
+import { ICanRenderProps } from './CanRenderBase';
 
-export const CanRender = (props: ICanRenderProps) => <CanRenderBase {...props} />;
+const CanRenderBase = React.lazy(() => import('./CanRenderBase'));
```

Comment:
This approach is necessary here? What do you think about to use the old import format?

Reply:
My idea was to only import this component when the component was rendering or in this case, when the condition passed to allow rendering were true but looking at a big picture, this is doing nothing there because the component that needs to be lazy-loaded is the component passed as children to CanRender, ElseCanRender, and ElseRender. So I will revert to the old approach.

Reply:
Solved.